src/vm/tests/contracts.rs

```diff
@@ -683,6 +683,19 @@ fn test_at_unknown_block() {
     with_marfed_environment(test, true);
 }
 
+#[test]
+fn test_as_max_len() {
+    fn test(owned_env: &mut OwnedEnvironment) {
+        let contract = "(define-data-var token-ids (list 10 uint) (list))
+                        (var-set token-ids 
+                           (unwrap! (as-max-len? (append (var-get token-ids) u1) u10) (err 10)))";
```

Comment:
Is there a way this function can test that `token-ids` now has length `u10`?

Also, it's not clear from the language documentation what should happen if the `M` value (`u10`) is less than `N` (the list's length) -- it returns `(none)` right?  There's no way to truncate a list?

Reply:
There's no way to truncate a list? My plan was to copy the elements to a shorter list.

Reply:
> Is there a way this function can test that token-ids now has length u10?

Yep— that’s what the assignment to data variable checks. 

> Also, it's not clear from the language documentation what should happen if the M value (u10) is less than N (the list's length) -- it returns (none) right?

Yes, it returns none. I can improve the description in the docs generation.

Reply:
And no, there’s no truncate or slice function at the moment. Would certainly be open to adding a native function for this, but it could be implemented with the fold function for now.

Reply:
@friedger want to open an issue to suggest adding a truncate-list built-in function?